src/index.js

```diff
@@ -21,42 +21,113 @@ import {
   constructor(opts = {}) {
-    this.settings = { ...DEFAULT_OPTIONS, ...opts };
+    this._settings = { ...DEFAULT_OPTIONS, ...opts };
     // a cache to store memoized srcset width-pairs
     this.targetWidthsCache = {};
-    if (typeof this.settings.domain != 'string') {
+    if (typeof this._settings.domain != 'string') {
       throw new Error('ImgixClient must be passed a valid string domain');
     }
 
-    if (DOMAIN_REGEX.exec(this.settings.domain) == null) {
+    if (DOMAIN_REGEX.exec(this._settings.domain) == null) {
       throw new Error(
         'Domain must be passed in as fully-qualified ' +
           'domain name and should not include a protocol or any path ' +
           'element, i.e. "example.imgix.net".',
       );
     }
 
-    if (this.settings.includeLibraryParam) {
-      this.settings.libraryParam = 'js-' + ImgixClient.version();
+    if (this._settings.includeLibraryParam) {
+      const defaultLibraryParam = 'js-' + ImgixClient.version();
+      this._settings.libraryParam =
+        this._settings.libraryParam || defaultLibraryParam;
     }
 
-    this.settings.urlPrefix = this.settings.useHTTPS ? 'https://' : 'http://';
+    this._settings.urlPrefix = this._settings.useHTTPS ? 'https://' : 'http://';
   }
 
   static version() {
     return VERSION;
   }
 
+  get settings() {
+    return this._settings;
+  }
+  get domain() {
+    return this._settings.domain;
+  }
+
+  set domain(value) {
+    if (typeof value != 'string') {
+      throw new Error('ImgixClient must be passed a valid string domain');
+    }
+
+    if (DOMAIN_REGEX.exec(value) == null) {
+      throw new Error(
+        'Domain must be passed in as fully-qualified ' +
+          'domain name and should not include a protocol or any path ' +
+          'element, i.e. "example.imgix.net".',
+      );
+    }
+    this._settings.domain = value;
+  }
+
+  get includeLibraryParam() {
+    return this._settings.includeLibraryParam;
+  }
+
+  set includeLibraryParam(value) {
+    if (typeof value !== 'boolean') {
+      throw 'includeLibraryParam must be a boolean';
+    }
+    this._settings.includeLibraryParam = value;
+  }
+
+  get libraryParam() {
+    return this._settings.libraryParam;
+  }
+
+  set libraryParam(value) {
+    if (typeof value !== 'string') {
+      throw 'libraryParam must be string';
+    }
+    this._settings.libraryParam = value;
+  }
+
+  get secureURLToken() {
+    return this._settings.secureURLToken;
+  }
+
+  set secureURLToken(value) {
+    if (typeof value === 'undefined') {
+      return (this._settings.secureURLToken = undefined);
+    }
+    if (typeof value !== 'string') {
+      throw 'secureURLToken must be string';
+    }
+    this._settings.secureURLToken = value;
+  }
+
+  get urlPrefix() {
+    return this._settings.urlPrefix;
+  }
+
+  set urlPrefix(value) {
+    if (typeof value !== 'string') {
+      throw 'urlPrefix must be string';
+    }
+    this._settings.urlPrefix = value;
+  }
+
```

Comment:
I did consider this. However, doing so required touching other functions that don't check for on/off but rather just for the prefix value.

Maybe the better option here was to define `get urlPrefix()` so that is uses `this.settings.useHTTPs` to determine the returned value.

``` js
  get urlPrefix() {
    if (this._settings.useHTTPs) {
      return "https://"
    }
    return "http://"
  }
```

This way, any call that references `this.urlPrefix` will always reflect the current value of `this.useHTTPs`

Reply:
Yep I agree with that approach